docs/alternator/compatibility.md

```diff
@@ -108,38 +108,45 @@ exactly the same microsecond-resolution timestamp, the the result may be
 a mixture of both writes - some attributes from one and some from the
 other - instead of being just one or the other.
 
-## Authorization
+## Authentication and Authorization
+
+By default, Alternator does not enforce authentication or authorization,
+and any request from any connected client will be allowed. To enforce
+authentication of the client, and authorization of which client is allowed
+to do what, configure the following in Scylla's configuration:
+
+    `alternator_enforce_authorization: true`
```

Comment:
I also am mentioning adding an extra Alternator config option in the [PRD](https://docs.google.com/document/d/1vzLcgYF4A3GdUuqbtXLDePYKlmd90FUbDqlrqAYjrME/edit?usp=sharing). Perhaps I should change PRD to state we'll use the already existing `alternator_enforce_authorization`?
Also, I find it a bit confusing that this option already exists and that the current documentation states:
> By default, authorization is not enforced at all. It can be turned on by providing an entry in Scylla configuration:
>     `alternator_enforce_authorization: true`

but only later the docs explain that we don't support authz.

Reply:
I look into this extra (or not extra) flag later.
At the moment, I saw the original "superuser" role has permissions to do anything, so this doesn't break. I assume (but will write a test later) that new superuser roles also has permissions to do anything. I somebody created a new non-superuser role but then expected it to behave like a superuser, well, indeed this user will have a problem. I'll consider how to handle this at the end, right now I want to implement the more important pieces of the functionality.

About the name "authorization" - in the past we made no distinction between authentication and authorization, so the result was this confusing flag name.

Reply:
> I look into this extra (or not extra) flag later.

Sure, I just wanted us to discuss it/keep it in mind rather than do something about it today.

> At the moment, I saw the original "superuser" role has permissions to do anything, so this doesn't break. I assume (but will write a test later) that new superuser roles also has permissions to do anything. I somebody created a new non-superuser role but then expected it to behave like a superuser, well, indeed this user will have a problem. I'll consider how to handle this at the end, right now I want to implement the more important pieces of the functionality.

Sounds good to me.

> About the name "authorization" - in the past we made no distinction between authentication and authorization, so the result was this confusing flag name.

Thanks for clarification.

Reply:
@nyh I'm returning to this question, because I saw Marcin approved the PR, but I wanted to discuss if this flag is needed to be handled before the PR gets merged, so are you planning to handle this flag in this patch or in a follow-up?
Generally I'm a little worried that checking permissions before executing any command will impact the performance of each Alternator's operation (perf impact needs to be measured), but if we first check if an authz flag is set to true, then only users which enabled authz will feel it.

Reply:
I'm not in a hurry to get this merged. I also agree that it needs more testing and performance testing, that it doesn't regress performance even when not used.